packages/cli/src/commands/devices/virtual/events.ts

```diff
@@ -0,0 +1,182 @@
+import {
+	DeviceEvent,
+	DeviceIntegrationType,
+} from '@smartthings/core-sdk'
+import {
+	APICommand,
+	inputAndOutputItem,
+	inputProcessor,
+	selectFromList,
+	TableGenerator,
+} from '@smartthings/cli-lib'
+import { VirtualDeviceEventsResponse } from '@smartthings/core-sdk/dist/endpoint/virtualdevices'
+import {
+	chooseAttribute,
+	chooseCapability,
+	chooseComponent,
+	chooseUnit,
+	chooseValue,
+} from '../../../lib/commands/devices/devices-util'
+
+
+function buildTableOutput(tableGenerator: TableGenerator, data: EventInputOutput): string {
+	const {input, output} = data
+	const table = tableGenerator.newOutputTable({head: ['Component','Capability','Attribute','Value','State Change?']})
+	const {stateChanges} = output
+	for (const index in input) {
+		const event = input[index]
+		const isStateChange = parseInt(index) < stateChanges.length ? stateChanges[index] : 'undefined'
+		table.push([event.component, event.capability, event.attribute, event.value, isStateChange])
+	}
+	return table.toString()
+}
+
+interface EventInputOutput {
+	input: DeviceEvent[]
+	output: VirtualDeviceEventsResponse
+}
+
+export default class VirtualDeviceEventsCommand extends APICommand<typeof VirtualDeviceEventsCommand.flags> {
+	static description = 'Create events for a virtual device\n' +
+		'The command can be run interactively, in question & answer mode, with command line parameters,\n' +
+		'or with input from a file or standard in.'
+
+	static examples = [
+		'$ smartthings devices:virtual:events                                                 # interactive mode',
+		'$ smartthings devices:virtual:events {id} -i data.yml                                # from a YAML or JSON file',
+		'$ smartthings devices:virtual:events {id} switch:switch on                           # command line input',
+		'$ smartthings devices:virtual:events {id} temperatureMeasurement:temperature 22.5 C  # command line input',
+	]
+
+	static flags = {
+		...APICommand.flags,
+		...inputAndOutputItem.flags,
+	}
+
+	static args = [
+		{
+			name: 'id',
+			description: 'the device id',
+		},
+		{
+			name: 'name',
+			description: 'the fully qualified attribute name [<component>]:<capability>:<attribute>',
+		},
+		{
+			name: 'value',
+			description: 'the attribute value',
+		},
+		{
+			name: 'unit',
+			description: 'optional unit of measure',
```

Comment:
Whether or not it is optional depends on the capability and attribute. For example, it is required for the `temperatureMeasurement` `temperature` attribute because there is no default temperature unit. It seems strange to make it a flag in that case since the command will fail if it is not provided.